llvm-external-projects/iree-dialects/include/iree-dialects/Dialect/LinalgExt/IR/LinalgExtOps.td

```diff
@@ -435,6 +435,61 @@ def IREELinalgExt_TopkOp : IREELinalgExt_Op<"topk",[
   }];
 }
 
+// TODO: is pack the right name? After all we are re-arranging the tensor.
+// TODO: destination passing style.
+def IREELinalgExt_PackOp : IREELinalgExt_Op<"pack",[
+  DeclareOpInterfaceMethods<LinalgExtInterface>,
+  DeclareOpInterfaceMethods<TilingInterface,
+    ["getIterationDomain",
+     "getLoopIteratorTypes",
+     "generateScalarImplementation"]>,
+  DeclareOpInterfaceMethods<MemoryEffectsOpInterface>
+]>{
+  let summary = "pack operation"; 
+
+  let arguments = (ins Variadic<AnyShaped>:$inputs,
+                       Variadic<AnyShaped>:$outputs,
+                       Variadic<Index>:$inner_tiles,
+                       I64ArrayAttr:$static_inner_tiles);
+
+  let results = (outs Variadic<AnyRankedTensor>:$results);
+  let assemblyFormat = [{
+    attr-dict
+    `ins` `(` $inputs `:` type($inputs) `)`
+    `outs` `(` $outputs `:` type($outputs) `)` 
```

Comment:
Hmm do we need this to be DPS ?
I am thinking of pack/unpack as similar to insert/extract ops that would produce new tensor values.
No need to change the current design if things already connect well, I'm just trying to think how we want things to look like in the fullness of time.

Reply:
Agree with you. At the tensor level should be 1 input, 1 result, 0 output. But I would like the op to work across tensor and memref for now (i.e., to see the loop generated). We can change later.